Cheza-Dzabala/way-farer
# Chore switch to postgres db

#### What does this PR do?

Removes all data structures from the application & implements a Postgresql Database.

#### Description of Task to be completed?

Removing all data structures from the application.  Replaced the datas tructures with 3 data bases:
 
Users
 ------------------ 
**id** | SERIAL INTEGER PRIMARY KEY
**first_name** | VARCHAR (128) NOT NULL
**last_name** | VARCHAR (128) NOT NULL
**email** | VARCHAR (128) NOT NULL
**password** | VARCHAR (128) NOT NULL


 Trips
 ------------------ 
**id** | SERIAL INTEGER PRIMARY KEY
**trip_date** | VARCHAR (128) NOT NULL
**bus_licesnse_number** | VARCHAR (128) NOT NULL
**origin** | VARCHAR (128) NOT NULL
**destination** | VARCHAR (128) NOT NULL
**fare** | VARCHAR (128) NOT NULL
**seating_capacity** | INTEGER NOT NULL
**number_of_seats** | INTEGENOT NULL


 Bookings
 ------------------ 
**id** | SERIAL INTEGER
**seat_number** | VARCHAR (128) NOT NULL
**user_id** | VARCHAR (128) NOT NULL
**trip_id** | VARCHAR (128) NOT NULL
PRIMARY KEY (user_id, trip_id)

#### How should this be manually tested?

1. Install PostgreSQL by following [these](https://www.postgresql.org/docs/9.3/tutorial-install.html) instructions.
2. Start PostgreSQL. 
3. Clone the repo. 
4. Install all dependencies by running ` npm install `.
5. Run the command: ` npm start `
6. Assuming you're on localhost:3000, sign up for account at ` localhost:3000/api/v2/auth/signup `
7. Check PostgreSQL for persistance. 
<-- opttional -->

To access protected routes:

1. Default user email: ` dzabalamacheza@gmail.com ` & password ` Runfree8418_!* `
2. To create your own admin, access route:
POST ` localhost:3000/api/v2/admins `

###### Entity Specification
```javascript
body: {
  "email": String,
   "first_name": String,
    "last_name": String,
    "password": String
}
````

###### Response Specification
```javascript
body: {
  "status": "success",
 "id": Int,
   "first_name": String,
    "last_name": String,
}
````

#### Any background context you want to provide?

Data was previously stored in Arrays

#### What are the relevant pivotal tracker stories?

###### #167826363
#### Screenshots (if appropriate)

<img width="1447" alt="Screenshot 2019-08-14 at 9 09 09 AM" src="https://user-images.githubusercontent.com/13886438/63001377-ceb51080-be73-11e9-925a-41eaccea11ca.png">

![screencapture-127-0-0-1-49164-browser-2019-08-14-09_09_20](https://user-images.githubusercontent.com/13886438/63001370-cc52b680-be73-11e9-9fd3-23be53f40d3f.png)


<img width="1460" alt="Screenshot 2019-08-14 at 9 08 32 AM" src="https://user-images.githubusercontent.com/13886438/63001378-cf4da700-be73-11e9-86a4-53a99de70d72.png">


<img width="1817" alt="Screenshot 2019-08-14 at 9 18 17 AM" src="https://user-images.githubusercontent.com/13886438/63002072-62d3a780-be75-11e9-90c3-3f8390fa3c56.png">

<img width="1461" alt="Screenshot 2019-08-14 at 9 07 41 AM" src="https://user-images.githubusercontent.com/13886438/63001379-cfe63d80-be73-11e9-9672-4b746bd26ecd.png">

![screencapture-127-0-0-1-49164-browser-2019-08-14-09_07_49](https://user-images.githubusercontent.com/13886438/63001373-cd83e380-be73-11e9-8898-e5c0f78d4ad9.png)

Updated Codebase
-------------------

<img width="1200" alt="Screenshot 2019-08-14 at 7 48 50 PM" src="https://user-images.githubusercontent.com/13886438/63043597-d654d380-becc-11e9-9339-894844adc22f.png">
<img width="985" alt="Screenshot 2019-08-14 at 7 48 37 PM" src="https://user-images.githubusercontent.com/13886438/63043599-d654d380-becc-11e9-986b-2df5db3978ba.png">
<img width="666" alt="Screenshot 2019-08-14 at 7 48 18 PM" src="https://user-images.githubusercontent.com/13886438/63043601-d654d380-becc-11e9-82e4-454145b6e81d.png">
<img width="537" alt="Screenshot 2019-08-14 at 7 48 06 PM" src="https://user-images.githubusercontent.com/13886438/63043602-d6ed6a00-becc-11e9-85ce-e2fa5501822d.png">
<img width="841" alt="Screenshot 2019-08-14 at 7 47 53 PM" src="https://user-images.githubusercontent.com/13886438/63043603-d6ed6a00-becc-11e9-8b3b-8d6da85bb9e6.png">
<img width="728" alt="Screenshot 2019-08-14 at 7 47 34 PM" src="https://user-images.githubusercontent.com/13886438/63043604-d6ed6a00-becc-11e9-8e2b-51d8e365162f.png">
<img width="527" alt="Screenshot 2019-08-14 at 7 47 24 PM" src="https://user-images.githubusercontent.com/13886438/63043605-d7860080-becc-11e9-870c-b9caf6c75baf.png">

#### Questions:

None
